api/src/main/java/io/druid/data/input/impl/TimestampSpec.java

```diff
@@ -40,6 +40,10 @@
   // this value should never be set for production data
   private final DateTime missingValue;
 
+  // remember last value parsed
+  private Object lastTimeObject = null;
+  private DateTime lastDateTime = null;
```

Comment:
Could you add to the class javadoc that this class is not thread safe? It used to be, and now it's not, but I think that's fine (no callers seem to be assuming thread safety). Let's make that explicit.